### PR TITLE
Add new character and weapon info from v5.1 Phase 1

### PR DESCRIPTION
**Pull request by sdglitched**
1. Add new character and weapon info and test from v5.1 Phase 1
- Character: Xilonen
Screenshot from 2024-11-05 21-43-47
Screenshot from 2024-11-05 21-43-57
- Weapons:
    - Sword:
        - Peak Patrol Song
        Screenshot from 2024-11-05 21-46-22
        Screenshot from 2024-11-05 21-46-29
        - Sturdy Bone
                Screenshot from 2024-11-05 21-48-26
                Screenshot from 2024-11-05 21-48-32
    - Polearm
        - Mountain-Bracing Bolt
        Screenshot from 2024-11-05 21-50-43
        Screenshot from 2024-11-05 21-50-49

2. Correct the order of SwordsDict
3. Update the supported Genshin Impact version information
Screenshot from 2024-11-05 21-52-09
Screenshot from 2024-11-05 21-52-15

Fixes: #210

### CODE REVIEW

File: assets/back.qrc [hunk 1]
```diff
@@ -80,6 +80,7 @@
     <file>imgs/char/back/xiangling.webp</file>
     <file>imgs/char/back/xianyun.webp</file>
     <file>imgs/char/back/xiao.webp</file>
+    <file>imgs/char/back/xilonen.webp</file>
```
Comment (gridhead) [inline assets/back.qrc:83]: Good work with the alphabetical order.

File: assets/cons.qrc [hunk 2]
```diff
@@ -78,6 +78,7 @@
     <file>imgs/char/cons/xiangling.webp</file>
     <file>imgs/char/cons/xianyun.webp</file>
     <file>imgs/char/cons/xiao.webp</file>
+    <file>imgs/char/cons/xilonen.webp</file>
```
Comment (gridhead) [inline assets/cons.qrc:81]: Good work with the alphabetical order.

File: assets/face.qrc [hunk 3]
```diff
@@ -80,6 +80,7 @@
     <file>imgs/char/face/xiangling.png</file>
     <file>imgs/char/face/xianyun.png</file>
     <file>imgs/char/face/xiao.png</file>
+    <file>imgs/char/face/xilonen.png</file>
```
Comment (gridhead) [inline assets/face.qrc:83]: Good work with the alphabetical order.

File: assets/name.qrc [hunk 4]
```diff
@@ -80,6 +80,7 @@
     <file>imgs/char/name/xiangling.png</file>
     <file>imgs/char/name/xianyun.png</file>
     <file>imgs/char/name/xiao.png</file>
+    <file>imgs/char/name/xilonen.png</file>
```
Comment (gridhead) [inline assets/name.qrc:83]: Good work with the alphabetical order.

File: gi_loadouts/data/weap/polearms/mtbb.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+        "Decreases Climbing Stamina Consumption by 15% and increases Elemental Skill DMG by 15%. Also, after other nearby party members use Elemental Skills, the equipping character's Elemental Skill DMG will also increase by 15% for 8s.",
+        "Decreases Climbing Stamina Consumption by 15% and increases Elemental Skill DMG by 18%. Also, after other nearby party members use Elemental Skills, the equipping character's Elemental Skill DMG will also increase by 18% for 8s.",
+        "Decreases Climbing Stamina Consumption by 15% and increases Elemental Skill DMG by 21%. Also, after other nearby party members use Elemental Skills, the equipping character's Elemental Skill DMG will also increase by 21% for 8s.",
+        "Decreases Climbing Stamina Consumption by 15% and increases Elemental Skill DMG by 24%. Also, after other nearby party members use Elemental Skills, the equipping character's Elemental Skill DMG will also increase by 24% for 8s.",
```
Comment (gridhead) [inline gi_loadouts/data/weap/polearms/mtbb.py:19]: 🌮

File: gi_loadouts/data/weap/swords/__init__.py [hunk 6]
```diff
@@ -23,6 +23,7 @@
     lofi,
     msrf,
     pmjc,
+    ppts,
```
Comment (gridhead) [inline gi_loadouts/data/weap/swords/__init__.py:26]: Good work with the alphabetical order.

File: gi_loadouts/data/weap/swords/__init__.py [hunk 7]
```diff
@@ -34,6 +35,7 @@
     spbd,
     srsd,
     swbd,
+    sybe,
```
Comment (gridhead) [inline gi_loadouts/data/weap/swords/__init__.py:38]: Good work with the alphabetical order.

File: gi_loadouts/data/weap/swords/__init__.py [hunk 8]
```diff
@@ -69,18 +71,20 @@
     "Light of Foliar Incision": lofi.LightofFoliarIncision,
     "Lion's Roar": lnrr.LionsRoar,
     "Mistsplitter Reforged": msrf.MistsplitterReforged,
+    "Peak Patrol Song": ppts.PeakPatrolSong,
```
Comment (gridhead) [inline gi_loadouts/data/weap/swords/__init__.py:74]: Good work with the alphabetical order.

File: gi_loadouts/data/weap/swords/__init__.py [hunk 9]
```diff
@@ -69,18 +71,20 @@
...
     "Primordial Jade Cutter": pmjc.PrimordialJadeCutter,
     "Prototype Rancour": ptrc.PrototypeRancour,
     "Royal Longsword": ryls.RoyalLongsword,
     "Sacrificial Sword": sfsd.SacrificialSword,
     "Sapwood Blade": spbd.SapwoodBlade,
     "Silver Sword": slsd.SilverSword,
+    "Skyrider Sword": srsd.SkyriderSword,
+    "Skyward Blade": swbd.SkywardBlade,
     "Splendor of Tranquil Waters": sotw.SplendorofTranquilWaters,
+    "Sturdy Bone": sybe.SturdyBone,
```
Comment (gridhead) [inline gi_loadouts/data/weap/swords/__init__.py:84]: Thank you for discovering this inconsistency and reporting the same on #210! 🎉

File: gi_loadouts/data/weap/swords/ppts.py [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+        "Gain \"Ode to Flowers\" after Normal or Plunging Attacks hit an opponent: DEF increases by 14% and gain a 17.5% All Elemental DMG Bonus for 6s. Max 2 stacks. Can trigger once per 0.1s. When this effect reaches 2 stacks or the 2nd stack's duration is refreshed, increase all nearby party members' All Elemental DMG Bonus by 14% for every 1,000 DEF the equipping character has, up to a maximum of 44.8%, for 15s.",
+        "Gain \"Ode to Flowers\" after Normal or Plunging Attacks hit an opponent: DEF increases by 16% and gain a 20% All Elemental DMG Bonus for 6s. Max 2 stacks. Can trigger once per 0.1s. When this effect reaches 2 stacks or the 2nd stack's duration is refreshed, increase all nearby party members' All Elemental DMG Bonus by 16% for every 1,000 DEF the equipping character has, up to a maximum of 51.2%, for 15s.",
```
Comment (gridhead) [inline gi_loadouts/data/weap/swords/ppts.py:19]: 🌮

File: gi_loadouts/data/weap/swords/ppts.py [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+from gi_loadouts.type.rare import Rare
+from gi_loadouts.type.weap import Sword, WeaponStat, WeaponStatType
+from gi_loadouts.type.weap.tier import Tier
+
+
+class PeakPatrolSong(Sword):
+    name: str = "Peak Patrol Song"
+    seco_stat: WeaponStat = WeaponStat(stat_name=WeaponStatType.defense_perc, stat_data=18.0)
+    tier: Tier = Tier.Tier_1
+    rare: Rare = Rare.Star_5
```
Comment (gridhead) [inline gi_loadouts/data/weap/swords/ppts.py:12]: 🌮

File: gi_loadouts/data/weap/swords/sybe.py [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+    refi_list: List[str] = [
+        "Sprint or Alternate Sprint Stamina Consumption decreased by 15%. Additionally, after using Sprint or Alternate Sprint, Normal Attack DMG is increased by 16% of ATK. This effect expires after triggering 18 times or 7s.",
+        "Sprint or Alternate Sprint Stamina Consumption decreased by 15%. Additionally, after using Sprint or Alternate Sprint, Normal Attack DMG is increased by 20% of ATK. This effect expires after triggering 18 times or 7s.",
+        "Sprint or Alternate Sprint Stamina Consumption decreased by 15%. Additionally, after using Sprint or Alternate Sprint, Normal Attack DMG is increased by 24% of ATK. This effect expires after triggering 18 times or 7s.",
+        "Sprint or Alternate Sprint Stamina Consumption decreased by 15%. Additionally, after using Sprint or Alternate Sprint, Normal Attack DMG is increased by 28% of ATK. This effect expires after triggering 18 times or 7s.",
+        "Sprint or Alternate Sprint Stamina Consumption decreased by 15%. Additionally, after using Sprint or Alternate Sprint, Normal Attack DMG is increased by 32% of ATK. This effect expires after triggering 18 times or 7s.",
+    ]
```
Comment (gridhead) [inline gi_loadouts/data/weap/swords/sybe.py:20]: 🌮

File: gi_loadouts/data/weap/swords/sybe.py [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+from gi_loadouts.type.rare import Rare
+from gi_loadouts.type.weap import Sword, WeaponStat, WeaponStatType
+from gi_loadouts.type.weap.tier import Tier
+
+
+class SturdyBone(Sword):
+    name: str = "Sturdy Bone"
+    seco_stat: WeaponStat = WeaponStat(stat_name=WeaponStatType.attack_perc, stat_data=6.0)
+    tier: Tier = Tier.Tier_3
+    rare: Rare = Rare.Star_4
```
Comment (gridhead) [inline gi_loadouts/data/weap/swords/sybe.py:12]: 🌮

File: test/data/weap/test_polearm.py [hunk 14]
```diff
@@ -35,6 +35,7 @@
         pytest.param("Kitain Cross Spear", 4, 3, "Level 80/90 (Rank 6)", 523, WeaponStatType.elemental_mastery, 101, id="data.weap.polearms: Kitain Cross Spear"),
         pytest.param("Lithic Spear", 4, 3, "Level 80/90 (Rank 6)", 523, WeaponStatType.attack_perc, 25.1, id="data.weap.polearms: Lithic Spear"),
         pytest.param("Moonpiercer", 4, 3, "Level 80/90 (Rank 6)", 523, WeaponStatType.elemental_mastery, 101, id="data.weap.polearms: Moonpiercer"),
+        pytest.param("Mountain-Bracing Bolt", 4, 3, "Level 80/90 (Rank 6)", 523, WeaponStatType.energy_recharge_perc, 27.9, id="data.weap.polearms: Mountain-Bracing Bolt"),
```
Comment (gridhead) [inline test/data/weap/test_polearm.py:38]: 🌮

File: test/data/weap/test_sword.py [hunk 15]
```diff
@@ -18,6 +18,7 @@
         pytest.param("Sacrificial Sword", 4, 1, "Level 80/90 (Rank 6)", 427, WeaponStatType.energy_recharge_perc, 55.9, id="data.weap.swords: Sacrificial Sword"),
         pytest.param("Key of Khaj-Nisut", 5, 1, "Level 80/90 (Rank 6)", 506, WeaponStatType.health_points_perc, 60.3, id="data.weap.swords: Key of Khaj-Nisut"),
         pytest.param("Light of Foliar Incision", 5, 1, "Level 80/90 (Rank 6)", 506, WeaponStatType.critical_damage_perc, 80.4, id="data.weap.swords: Light of Foliar Incision"),
+        pytest.param("Peak Patrol Song", 5, 1, "Level 80/90 (Rank 6)", 506, WeaponStatType.defense_perc, 75.4, id="data.weap.swords: Peak Patrol Song"),
```
Comment (gridhead) [inline test/data/weap/test_sword.py:21]: 🌮

File: test/data/weap/test_sword.py [hunk 16]
```diff
@@ -49,6 +50,7 @@
         pytest.param("Finale of the Deep", 4, 3, "Level 80/90 (Rank 6)", 523, WeaponStatType.attack_perc, 25.1, id="data.weap.swords: Finale of the Deep"),
         pytest.param("Prototype Rancour", 4, 3, "Level 80/90 (Rank 6)", 523, WeaponStatType.damage_bonus_physical_perc, 31.5, id="data.weap.swords: Prototype Rancour"),
         pytest.param("Sapwood Blade", 4, 3, "Level 80/90 (Rank 6)", 523, WeaponStatType.energy_recharge_perc, 27.9, id="data.weap.swords: Sapwood Blade"),
+        pytest.param("Sturdy Bone", 4, 3, "Level 80/90 (Rank 6)", 523, WeaponStatType.attack_perc, 25.1, id="data.weap.swords: Sturdy Bone"),
```
Comment (gridhead) [inline test/data/weap/test_sword.py:53]: 🌮

File: test/face/wind/weap/kind/test_polearm.py [hunk 17]
```diff
@@ -35,6 +35,7 @@
         pytest.param("Kitain Cross Spear", 4, "Level 80/90 (Rank 6)", 523, WeaponStatType.elemental_mastery, 101, 5, id="face.wind.rule: Configuring weapon - Polearm - Kitain Cross Spear"),
         pytest.param("Lithic Spear", 4, "Level 80/90 (Rank 6)", 523, WeaponStatType.attack_perc, 25.1, 5, id="face.wind.rule: Configuring weapon - Polearm - Lithic Spear"),
         pytest.param("Moonpiercer", 4, "Level 80/90 (Rank 6)", 523, WeaponStatType.elemental_mastery, 101, 5, id="face.wind.rule: Configuring weapon - Polearm - Moonpiercer"),
+        pytest.param("Mountain-Bracing Bolt", 4, "Level 80/90 (Rank 6)", 523, WeaponStatType.energy_recharge_perc, 27.9, 5, id="face.wind.rule: Configuring weapon - Polearm - Mountain-Bracing Bolt"),
```
Comment (gridhead) [inline test/face/wind/weap/kind/test_polearm.py:38]: 🌮

File: test/face/wind/weap/kind/test_sword.py [hunk 18]
```diff
@@ -18,6 +18,7 @@
         pytest.param("Sacrificial Sword", 4, "Level 80/90 (Rank 6)", 427, WeaponStatType.energy_recharge_perc, 55.9, 5, id="face.wind.rule: Configuring weapon - Sword - Sacrificial Sword"),
         pytest.param("Key of Khaj-Nisut", 5, "Level 80/90 (Rank 6)", 506, WeaponStatType.health_points_perc, 60.3, 5, id="face.wind.rule: Configuring weapon - Sword - Key of Khaj-Nisut"),
         pytest.param("Light of Foliar Incision", 5, "Level 80/90 (Rank 6)", 506, WeaponStatType.critical_damage_perc, 80.4, 5, id="face.wind.rule: Configuring weapon - Sword - Light of Foliar Incision"),
+        pytest.param("Peak Patrol Song", 5, "Level 80/90 (Rank 6)", 506, WeaponStatType.defense_perc, 75.4, 5, id="face.wind.rule: Configuring weapon - Sword - Peak Patrol Song"),
```
Comment (gridhead) [inline test/face/wind/weap/kind/test_sword.py:21]: 🌮

File: test/face/wind/weap/kind/test_sword.py [hunk 19]
```diff
@@ -49,6 +50,7 @@
         pytest.param("Finale of the Deep", 4, "Level 80/90 (Rank 6)", 523, WeaponStatType.attack_perc, 25.1, 5, id="face.wind.rule: Configuring weapon - Sword - Finale of the Deep"),
         pytest.param("Prototype Rancour", 4, "Level 80/90 (Rank 6)", 523, WeaponStatType.damage_bonus_physical_perc, 31.5, 5, id="face.wind.rule: Configuring weapon - Sword - Prototype Rancour"),
         pytest.param("Sapwood Blade", 4, "Level 80/90 (Rank 6)", 523, WeaponStatType.energy_recharge_perc, 27.9, 5, id="face.wind.rule: Configuring weapon - Sword - Sapwood Blade"),
+        pytest.param("Sturdy Bone", 4, "Level 80/90 (Rank 6)", 523, WeaponStatType.attack_perc, 25.1, 5, id="face.wind.rule: Configuring weapon - Sword - Sturdy Bone"),
```
Comment (gridhead) [inline test/face/wind/weap/kind/test_sword.py:53]: 🌮

**Review by gridhead — APPROVED** [review]
Looks good to me! Thanks for working on this 🎉 

One small nitpick would be to keep the functional codebase and the test cases in separate commits.

**Comment by Ank-22** [comment]
I have cross-checked the added characters and weapons; all presents are bug-free.

**Comment by AvadhootDhere** [comment]
Everything looks good as always, ready for merge

**Pull request merged**